perception/traffic_light_visualization/src/traffic_light_roi_visualizer/shape_draw.cpp

```diff
@@ -0,0 +1,158 @@
+// Copyright 2024 The Autoware Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include "traffic_light_visualization/traffic_light_roi_visualizer/shape_draw.hpp"
+
+#include "opencv2/core/types.hpp"
+#include "opencv2/highgui.hpp"
+#include "opencv2/imgproc.hpp"
+
+void drawShape(
+  const DrawFunctionParams & params, const std::string & filename, bool flipHorizontally,
+  bool flipVertically, int x_offset, int y_offset)
+{
+  std::string filepath =
+    ament_index_cpp::get_package_share_directory("traffic_light_visualization") + "/images/" +
+    filename;
+  cv::Mat shapeImg = cv::imread(filepath, cv::IMREAD_UNCHANGED);
+  if (shapeImg.empty()) {
+    std::cerr << "Failed to load image: " << filepath << std::endl;
+    return;
+  }
+
+  if (flipHorizontally) {
+    cv::flip(shapeImg, shapeImg, 1);  // Flip horizontally
+  }
+
+  if (flipVertically) {
+    cv::flip(shapeImg, shapeImg, 0);  // Flip vertically
+  }
+
+  // Resize image if needed
+  double scale_factor = 0.25;  // Scale factor to reduce the size by 50%
```

Comment:
Could you please make scale_factor a parameter instead of hard-coded?

Reply:
Done